PWGUD/Tasks/upcJpsiCentralBarrelCorr.cxx

```diff
@@ -0,0 +1,888 @@
+        }
+      }
+
+      RawData.get<TH1>(HIST("RawData/hTrackPt"))->Fill(track.pt());
+      RawData.get<TH1>(HIST("RawData/hTrackEta"))->Fill(RecoDecay::eta(std::array{trkPx, trkPy, trkPz}));
+      RawData.get<TH1>(HIST("RawData/hTrackPhi"))->Fill(RecoDecay::phi(trkPx, trkPy));
+
+      if (track.hasTPC()) {
+        RawData.get<TH2>(HIST("RawData/PID/hTPCVsP"))->Fill(RecoDecay::sqrtSumOfSquares(trkPx, trkPy, trkPz), track.tpcSignal());
+        RawData.get<TH2>(HIST("RawData/PID/hTPCVsPt"))->Fill(track.pt(), track.tpcSignal());
+        RawData.get<TH2>(HIST("RawData/PID/hTPCVsEta"))->Fill(RecoDecay::eta(std::array{trkPx, trkPy, trkPz}), track.tpcSignal());
+        RawData.get<TH2>(HIST("RawData/PID/hTPCVsPhi"))->Fill(RecoDecay::phi(trkPx, trkPy), track.tpcSignal());
+      }
+
+      if (track.hasTOF()) {
+        RawData.get<TH2>(HIST("RawData/PID/hTOFVsP"))->Fill(RecoDecay::sqrtSumOfSquares(trkPx, trkPy, trkPz), track.tofSignal());
+        RawData.get<TH2>(HIST("RawData/PID/hTOFVsPt"))->Fill(track.pt(), track.tofSignal());
+        RawData.get<TH2>(HIST("RawData/PID/hTOFVsEta"))->Fill(RecoDecay::eta(std::array{trkPx, trkPy, trkPz}), track.tofSignal());
+        RawData.get<TH2>(HIST("RawData/PID/hTOFVsPhi"))->Fill(RecoDecay::phi(trkPx, trkPy), track.tofSignal());
+      }
+    }
+
+    int countGT = 0;
+    int countGTselected = 0;
+    int countGTel = 0;
+    int countGTmu = 0;
+    int countGTp = 0;
+    std::vector<int> trkIdx;
+    // loop over tracks with selections
+    for (auto& track : tracks) {
+      // select primary vertex contributors
+      if (track.isPVContributor() != 1) {
+        return;
+      }
+      // select good tracks
+      if (GoodTrackCuts(track) != 1) {
+        return;
+      }
+      countGT++;
+      int hypoID = testPIDhypo(track);
+      if (hypoID == P_ELECTRON || hypoID == P_MUON) {
+        countGTselected++;
+        trkIdx.push_back(track.index());
+        if (hypoID == P_ELECTRON) {
+          countGTel++;
+        }
+        if (hypoID == P_MUON) {
+          countGTmu++;
+        }
+        if (hypoID == P_PROTON) {
+          countGTp++;
+        }
+      }
+    }
+
+    Statistics.get<TH1>(HIST("Statistics/hNumberOfTracks"))->Fill(2., countGT);
+    Statistics.get<TH1>(HIST("Statistics/hNumberOfTracks"))->Fill(3., countGTselected);
+    Statistics.get<TH1>(HIST("Statistics/hNumberOfTracks"))->Fill(4., countGTel);
+    Statistics.get<TH1>(HIST("Statistics/hNumberOfTracks"))->Fill(5., countGTmu);
+    Statistics.get<TH1>(HIST("Statistics/hNumberGT"))->Fill(countGT);
+    Statistics.get<TH1>(HIST("Statistics/hNumberGTselected"))->Fill(countGTselected);
+    Statistics.get<TH1>(HIST("Statistics/hNumberGTel"))->Fill(countGTel);
+    Statistics.get<TH1>(HIST("Statistics/hNumberGTmu"))->Fill(countGTmu);
+    Statistics.get<TH1>(HIST("Statistics/hNumberGTp"))->Fill(countGTp);
+
+    float massEl = 0.000510998950;
+    float massMu = 0.10565837;
+    float massPr = 0.93827208816;
```

Comment:
As I said, you should remove the `particleMass` function and get the masses from the physics constants header.
`#include "CommonConstants/PhysicsConstants.h"`